Migrates workbox module docs from WebFu to developer.chrome.com

malchata opened the pull request:
Changes proposed in this pull request:
- Migrates Workbox module documentation from WebFundamentals to developer.chrome.com

Notes:
- I couldn't find a way to nest the modules within the `/docs/workbox/` folder, so this now lives at `/docs/workbox-modules/`. Feel free to change this up if need be, but if we do, I'll need to update internal links to Workbox modules (not really a big deal).
- I've done my best to ensure all links are working. There should be no 404s, but some links will end up going to the old reference docs until we get them moved over. That said, feel free to audit links you know are wrong/should be updated.
- Some reformatting was necessary to remove old WebFu components.
- Any paragraphs prefaced with "Note:" were changed to the web.dev `{% Aside %}` component.
- Any paragraphs prefaced with "Warning:" were changed to the web.dev `{% Aside 'warning' %}` component.
- Any paragraphs prefaced with "Caution:" were changed to the web.dev `{% Aside 'caution' %}` component.
- Table markup (where applicable) was reformatted to use the web.dev table component markup.

I am 100% sure I have missed some stuff, so a thorough review would be appreciated. I expect we'll have to modify these again when we land the next wave of docs.

jeffposnick commented on site/en/docs/workbox-modules/workbox-broadcast-update/index.md:
```diff
@@ -0,0 +1,168 @@
+date: 2017-11-29
+updated: 2020-05-01
+description: >
+  Send messages to pages (via Broadcast Channels) when a cache is updated with a new response.
```
```suggestion
  Send messages to pages when a cache is updated with a new response.
```

jeffposnick commented on site/en/docs/workbox-modules/workbox-build/index.md:
```diff
@@ -0,0 +1,119 @@
+date: 2018-01-31
+updated: 2020-01-17
+description: >
+  An npm module that can generate a complete service worker, inject a precache manifest and copy the Workbox files.
```
```suggestion
  A module that can generate a service worker, inject a precache manifest, or create a local copy the Workbox libraries.
```

jeffposnick commented on site/en/docs/workbox-modules/workbox-cli/index.md:
```diff
@@ -0,0 +1,183 @@
+date: 2017-11-27
+updated: 2020-01-17
+description: >
+  Generate a complete service worker, inject a precache manifest or copy the Workbox files from the command line.
```
```suggestion
  Generate a service worker, inject a precache manifest, or create a local copy the Workbox libraries from the command line.
```

jeffposnick commented on site/en/docs/workbox-modules/workbox-google-analytics/index.md:
```diff
@@ -0,0 +1,183 @@
+date: 2017-11-27
+updated: 2021-04-23
+description: >
+  See how users are interacting with your site when they are offline.
```
```suggestion
  Support for replaying offline Google Analytics interactions.
```

jeffposnick commented on site/en/docs/workbox-modules/workbox-sw/index.md:
```diff
@@ -0,0 +1,231 @@
+date: 2017-11-27
+updated: 2021-08-23
+description: >
+  Manages dynamic loading of Workbox service worker packages as needed and provides additional helper methods.
```
```suggestion
  Provides a way to load the Workbox runtime without a build step.
```

jeffposnick commented on site/en/docs/workbox-modules/workbox-webpack-plugin/index.md:
```diff
@@ -0,0 +1,147 @@
+date: 2017-12-15
+updated: 2020-02-03
+description: >
+  As part of your Webpack build process add Workbox and precache assets.
```
```suggestion
  Generate a service worker or inject a precache manifest, using the webpack build tool.
```

jeffposnick reviewed (changes requested):
Thanks, @malchata!

I've added a few inline suggestions to clean up imprecise language we used on the old landing pages.

I am curious as to what problem you had nesting these under a `/docs//workbox/modules/*` path, though. I think that, generally speaking, our relative URLs will end up transitioning over from d.g.c. to d.c.c. with fewer required changes if we can keep the same directory structure. Maybe it's something that the d.c.c. infra team could help with if there's a technical blocker?

malchata commented:
> I am curious as to what problem you had nesting these under a `/docs/workbox/modules/*` path, though. I think that, generally speaking, our relative URLs will end up transitioning over from d.g.c. to d.c.c. with fewer required changes if we can keep the same directory structure. Maybe it's something that the d.c.c. infra team could help with if there's a technical blocker?

It simply wasn't clear to me how the nesting of one docs folder under another would work. I had attempted a few different things, but wasn't really getting anywhere. In the interest of moving forward, I just decided to do what I know would work for now and pivot during the PR phase if need be.

malchata commented on site/en/docs/workbox-modules/workbox-routing/index.md:
```diff
@@ -0,0 +1,291 @@
+this regular expression will match as long as the request's URL matches the
+regular expression.
+
+- https://example.com**1">/styles/main.css**
+- https://example.com**1">/styles/nested/file.css**
```
```suggestion
- https://example.com**/styles/main.css**
- https://example.com**/styles/nested/file.css**
```

malchata commented:
@jeffposnick, with @MichaelSolati's help, the workbox module docs now reside under /docs/workbox/modules. All internal links to to module docs were also updated, as well as direct links to specific sections.

Additionally, I updated internal links to the reference docs where possible, but could not do this for every module (e.g., `workbox-build`, `workbox-webpack-plugin`, etc.) since not all modules have an API reference generated for them.

Unless you have additional feedback, I think we should merge this in anticipation of the next wave of docs and updates to 
the API reference so we can wrap this all up!

jeffposnick reviewed (approved):
Thanks for getting the URL structure sorted!

I think we have to think about navigation/table of content things at some point, to figure out how to surface these module docs in the overall site navigation. But for the time being, everything in this PR looks good.
